webui/react/src/pages/TrialDetails/useTrialMetrics.ts

```diff
@@ -0,0 +1,104 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { Serie } from 'components/kit/LineChart';
+import { XAxisDomain } from 'components/kit/LineChart/XAxisFilter';
+import { terminalRunStates } from 'constants/states';
+import useMetricNames from 'hooks/useMetricNames';
+import { compareTrials } from 'services/api';
+import usePolling from 'shared/hooks/usePolling';
+import { isEqual } from 'shared/utils/data';
+import { ErrorType } from 'shared/utils/error';
+import { Metric, MetricContainer, RunState, TrialDetails } from 'types';
+import handleError from 'utils/error';
+import { metricToKey } from 'utils/metric';
+
+type MetricName = string;
+
+export interface TrialMetrics {
+  data: Record<MetricName, Serie>;
+  metrics: Metric[];
+}
+
+const summarizedMetricToSeries = (summ: MetricContainer): Serie => {
+  const rawData: [number, number][] = [];
+  const rawTime: [number, number][] = [];
+  summ.data.forEach((dataPoint) => {
+    rawData.push([dataPoint.batches, dataPoint.value]);
+  });
+
+  summ.time?.forEach((dataPoint) => {
+    rawTime.push([new Date(dataPoint.time).getTime() / 1000, dataPoint.value]);
+  });
+
+  const data: Partial<Record<XAxisDomain, [number, number][]>> = {
+    [XAxisDomain.Batches]: rawData,
+    [XAxisDomain.Time]: rawTime,
+  };
+
+  return {
+    data,
+    metricType: summ.type,
+    name: summ.name,
+  };
+};
+
+export const useTrialMetrics = (
+  trial: TrialDetails | undefined,
+): { data: Record<string, Serie> | undefined; metrics: Metric[] } => {
+  const trialTerminated = terminalRunStates.has(trial?.state ?? RunState.Active);
+
+  const handleMetricNamesError = useCallback(
+    (e: unknown) => {
+      handleError(e, {
+        publicMessage: `Failed to load metric names for trial ${trial?.id}.`,
+        publicSubject: 'Experiment metric name stream failed.',
+        type: ErrorType.Api,
+      });
+    },
+    [trial?.id],
+  );
+
+  const metrics = useMetricNames(trial?.experimentId, handleMetricNamesError);
+  const [data, setData] = useState<Record<MetricName, Serie>>();
+
+  const fetchTrialSummary = useCallback(async () => {
+    if (trial?.id) {
+      const response = await compareTrials({
+        maxDatapoints: screen.width > 1600 ? 1500 : 1000,
+        metricNames: metrics,
+        // scale: scale,
```

Comment:
intended to comment out?

Reply:
The state for the scale is entirely within the component, so we can't know it from outside. 

In any case, the scale parameter does not affect the correctness of the data being passed back, just the sampling. 

Worth revisiting, some possibilities:
- Make the ChartGrid component a "dumb component" with axes as state that is external and with external handlers (the main pattern in our codebase)
- call API with both values for the scale parameter and put it into a dict (or have the API just return it both way). Not a fan of this option.
- leave as is (maybe non-logarithmic downsampling is sufficient)

Reply:
I could also see passing some onScaleChange={callback} to the ChartGrid.  But we can make resolving this into its own ticket

Reply:
that's what I meant by this:
> axes as state that is external and with external handlers
i think we should hold off till making the scales chart-specific though.